### DISCUSSION TITLE
ReadModelPopulator.PopulateAsync returning prematurely.

### DISCUSSION THREAD

**Discussion by IanEsteves5** [rasmus/EventFlow#863] [Q&A]
Hello.

I am trying to use the **ReadModelPopulator** class to repopulate my read model tables in my database. I ran into an issue where the populator would just say that no relevant events were found and leave the table empty, even though I knew for a fact that it should have found them.

> Population of read model 'ExampleReadModel' took 0,298 seconds, in which 5 events was loaded and 0 was relevant

I then tried setting the **EventFlowConfiguration.PopulateReadModelEventPageSize** to a very large number, and the read models were upraded succesfully.

 ```csharp
EventFlowOptions.New.Configure(c => c.PopulateReadModelEventPageSize = 40000)
```

> Population of read model 'ExampleReadModel' took 23,4 seconds, in which 80819 events was loaded and 32 was relevant

I'm not sure what is causing this issue. Maybe it has to do with event upgraders? There are a few of them in my code. Another thing I noticed is that there is a pretty big gap in the **GlobalSequenceNumber** column in the **EventFlow** table (about 3500).

Thank you,
                      Ian.

**Comment by KoriSeng** [comment]
Not sure if this helps.

https://github.com/eventflow/EventFlow/issues/666#issuecomment-512628983
